lib/igniter/code/function.ex

```diff
@@ -24,35 +25,98 @@ defmodule Igniter.Code.Function do
     end
   end
 
-  @spec move_to_defp(Zipper.t(), fun :: atom, arity :: integer | list(integer)) ::
+  @spec move_to_defp(Zipper.t(), fun :: atom, arity :: integer | list(integer), Keyword.t()) ::
           {:ok, Zipper.t()} | :error
-  def move_to_defp(zipper, fun, arity) do
-    do_move_to_def(zipper, fun, arity, :defp)
+        when opts: {:target, :inside | :before | :at}
+  def move_to_defp(zipper, fun, arity, opts \\ [target: :inside]) do
+    do_move_to_def(zipper, fun, arity, :defp, opts)
   end
 
-  @spec move_to_def(Zipper.t()) :: {:ok, Zipper.t()} | :error
-  def move_to_def(zipper) do
-    move_to_function_call(zipper, :def, :any)
+  @spec move_to_def(Zipper.t(), Keyword.t()) :: {:ok, Zipper.t()} | :error
+        when opts: {:target, :inside | :before | :at}
+  def move_to_def(zipper, opts \\ [target: :at]) do
+    case move_to_function_call(zipper, :def, :any) do
+      {:ok, zipper} ->
+        move_to_target(zipper, Keyword.get(opts, :target, :inside))
+
+      :error ->
+        :error
+    end
   end
 
-  @spec move_to_def(Zipper.t(), fun :: atom, arity :: integer | list(integer)) ::
+  @doc """
+  Moves the zipper to a function definition by the given name and arity. You may
+  also pass in a :target option to specify where in the function you want to
+  move to. By default it will move to the inside of the function.
+  The `:target` option can be one of the following:
+  - `:inside` - moves to the inside of the function
+  - `:before` - moves to before the function and takes into consideration the
+    attributes `@doc`, `@spec`, and `@impl` if they exist
+  - `:at` - moves to the function definition itself. Use this if you want to add
+    code directly before or directly after the function.
+
+  ## Example - Moves before the function.
+
+  ```elixir
+  zipper =
+    \"\"\"
+    defmodule Test do
+      @doc "hello"
+      @spec hello() :: :world
+      def hello() do
+        :world
+      end
+    end
+    \"\"\"
+    |> Sourceror.parse_string!()
+    |> Zipper.zip()
+
+  {:ok, zipper} = Igniter.Code.Function.move_to_function_and_attrs(zipper, :hello, 0)
+
+  zipper =
+    Igniter.Code.Common.add_code(
+      zipper,
+      \"\"\"
+      def world() do
+        :hello
+      end
+      \"\"\",
+      placement: :before
+    )
+
+  Igniter.Util.Debug.code_at_node(Zipper.topmost(zipper))
+  # defmodule Test do
+  #   def world() do
+  #     :hello
+  #   end
+  #
+  #   @doc "hello"
+  #   @spec hello() :: :world
+  #   def hello() do
+  #     :world
+  #   end
+  # end
+  ```
+  """
+  @spec move_to_def(Zipper.t(), fun :: atom, arity :: integer | list(integer), Keyword.t()) ::
           {:ok, Zipper.t()} | :error
-  def move_to_def(zipper, fun, arity) do
-    do_move_to_def(zipper, fun, arity, :def)
+        when opts: {:target, :inside | :before | :at}
+  def move_to_def(zipper, fun, arity, opts \\ [target: :inside]) do
```

Comment:
instead of defaulting the `opts` to a value, we should default to an empty list and use tools like `Keyword.get(opts, :target, :inside)` to access the default value.